hydra-github-bridge/app/Main.hs

```diff
@@ -181,25 +189,54 @@ mkStatus :: Maybe Text -> Maybe Text -> Maybe Text -> Maybe Text -> Text -> Text
 
 mkStatus = client (Proxy @GitHubAPI)
 
-statusHandler :: Text -> TChan GitHubStatus -> IO ()
-statusHandler token queue = do
-    action <- atomically (readTChan queue)
-    print action
+-- TODO: What do we do if the request fails?
+sendStatusToGitHub :: Text -> GitHubStatus -> IO ()
+sendStatusToGitHub token status = do
     manager <- newManager tlsManagerSettings
     let env = (mkClientEnv manager (BaseUrl Https "api.github.com" 443 ""))
-    putStrLn $ BSL.unpack $ encode (payload action)
-    res <- flip runClientM env $ do
+    putStrLn $ BSL.unpack $ encode (payload status)
+    void . flip runClientM env $ do
         mkStatus (Just "hydra-github-bridge")
                  (Just "application/vnd.github+json")
                  (Just token)
                  (Just "2022-11-28")
-                 (owner action)
-                 (repo action)
-                 (sha action)
-                 (payload action)
-    print res
-    -- todo make servant client request
+                 (owner status)
+                 (repo status)
+                 (sha status)
+                 (payload status)
+
+saveStatusToDb :: Connection -> GitHubStatus -> IO Int
+saveStatusToDb conn status = do
+    let q = "insert into github_status (owner, repo, sha, payload) values (?, ?, ?, ?) returning id"
+    [Only id'] <- query conn q (owner status, repo status, sha status, encode (payload status))
+    pure id'
 
+lookupStatusFromDB :: Connection -> Int -> IO (Maybe GitHubStatus)
+lookupStatusFromDB conn id' = do
+    let q = "select owner, repo, sha, payload from github_status where id = ?"
+    [(owner', repo', sha', payload')] <- query conn q (Only id')
+
+    case fromJSON payload' of
+        Aeson.Success p -> do
+          pure $ Just (GitHubStatus owner' repo' sha' p)
+        Aeson.Error err -> do
+          Text.putStrLn $ "Could not parse status: " <> toStrict (encodeToLazyText payload')
+          pure Nothing
+
+-- For each new status, run the procedure:
+--
+--  1. Save the status to a Hydra table
+--  2. Send it to GitHub
+--  3. Delete it from the Hydra table
+--
+-- TODO: What do we do with statuses left in the database?
+handleStatus :: Connection -> Text -> GitHubStatus -> IO ()
+handleStatus conn token status = do
+    id' <- saveStatusToDb conn status
+    void . forkIO $
+      lookupStatusFromDB conn id' >>= \case
+        Just status -> sendStatusToGitHub token status
+        Nothing -> pure ()
```

Comment:
As discussed we probably want to split this into two jobs.
1. sticking the stuff into the db (as we do right now)
2. have a separate thread work down the notifications to be sent to GitHub. We can have a LISTEN / NOTIFY logic, e.g. have the sender thread LISTEN on "notification_added", and the notification writer thread write a "NOTIFY" message after writing the status to the DB. (Or we create a trigger that creates the NOTIFY whenever items were added to the table).

We probably also want to have a "sent" timestamp field (I'll adjust the schema). And UPDATE that on successful notification delivery. We can then insert a trigger into the DB to DELETE anything that has been sent more than 5 days ago.